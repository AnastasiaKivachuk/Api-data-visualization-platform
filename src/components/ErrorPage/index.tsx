import React, {FC} from 'react';
import Link from 'next/link';

import Button from "@mui/material/Button";
import Typography from "@mui/material/Typography";

import {HOME_ROUTE} from "../../constants/routes.constants";

interface Props {
    error?: string
    text?: string
}

const ErrorPage: FC<Props> = ({error, text}) => (
    <>
        <Typography variant="h1" textAlign='center'>
            {error}
        </Typography>
        <Typography variant="h3" textAlign='center'>
            Ooops, something went wrong
        </Typography>
        <Typography variant="subtitle1" textAlign='center'>
            {text}
        </Typography>
        <Link href={HOME_ROUTE}>
            <Button variant={'contained'}>Go to the main page</Button>
        </Link>
    </>
);

ErrorPage.defaultProps = {
    error: '404',
    text: 'The page you are looking for doesn\'t exist or an other error occurred. Go back, or head over to the main page.',
};

export default ErrorPage;
