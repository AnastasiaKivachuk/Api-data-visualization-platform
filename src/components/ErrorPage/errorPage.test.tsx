import {shallow} from "enzyme";

import ErrorPage from "./index";


const MockErrorData = {
    error: "500",
    text: "The server encountered an internal error or misconfiguration. Please, refresh this page or try again later."
}

const setUp = (props: { error: string, text: string } | {}) => shallow(<ErrorPage {...props}/>);

describe("ErrorPage component", () => {

    it("should render ErrorPage component with default Props", () => {
        const component = setUp({});
        expect(component).toMatchSnapshot();
    })

    it("should render ErrorPage component with props", () => {
        const component = setUp(MockErrorData);
        expect(component).toMatchSnapshot();
    })
})
