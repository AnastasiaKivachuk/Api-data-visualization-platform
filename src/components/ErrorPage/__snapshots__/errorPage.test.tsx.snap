// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage component should render ErrorPage component with default Props 1`] = `
<Fragment>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h1"
  >
    404
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h3"
  >
    Ooops, something went wrong
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="subtitle1"
  >
    The page you are looking for doesn't exist or an other error occurred. Go back, or head over to the main page.
  </ForwardRef(Typography)>
  <ForwardRef
    href="/"
  >
    <ForwardRef(Button)
      variant="contained"
    >
      Go to the main page
    </ForwardRef(Button)>
  </ForwardRef>
</Fragment>
`;

exports[`ErrorPage component should render ErrorPage component with props 1`] = `
<Fragment>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h1"
  >
    500
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="h3"
  >
    Ooops, something went wrong
  </ForwardRef(Typography)>
  <ForwardRef(Typography)
    textAlign="center"
    variant="subtitle1"
  >
    The server encountered an internal error or misconfiguration. Please, refresh this page or try again later.
  </ForwardRef(Typography)>
  <ForwardRef
    href="/"
  >
    <ForwardRef(Button)
      variant="contained"
    >
      Go to the main page
    </ForwardRef(Button)>
  </ForwardRef>
</Fragment>
`;
